src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -109,42 +112,61 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
        shapes_it != tasks_to_dispatch_.end();) {
     auto &dispatch_queue = shapes_it->second;
     for (auto work_it = dispatch_queue.begin(); work_it != dispatch_queue.end();) {
-      auto work = *work_it;
-      auto task = std::get<0>(work);
-      auto spec = task.GetTaskSpecification();
+      auto &work = *work_it;
+      auto &task = std::get<0>(work);
+      auto &spec = task.GetTaskSpecification();
 
       std::shared_ptr<WorkerInterface> worker = worker_pool.PopWorker(spec);
       if (!worker) {
         // No worker available, we won't be able to schedule any kind of task.
         return;
       }
 
-      std::shared_ptr<TaskResourceInstances> allocated_instances(
-          new TaskResourceInstances());
-      bool schedulable = cluster_resource_scheduler_->AllocateLocalTaskResources(
-          spec.GetRequiredResources().GetResourceMap(), allocated_instances);
-      if (!schedulable) {
-        // Not enough resources to schedule this task.
+      const auto owner_worker_id = WorkerID::FromBinary(spec.CallerAddress().worker_id());
+      const auto owner_node_id = NodeID::FromBinary(spec.CallerAddress().raylet_id());
+      // If the owner has died since this task was queued, cancel the task by
+      // killing the worker (unless this task is for a detached actor).
+      if (!worker->IsDetachedActor() &&
```

Comment:
I don't think the check for the worker being a detached actor makes sense here since the worker shouldn't be executing anything yet (the other place where this check shows up is for already leased workers). I think maybe we should check if the task spec is a detached actor creation task, though. Do we have a Python test for the driver exiting without waiting for a detached actor to be created?

Reply:
Good call (I think this might be buggy in the current scheduler now 🤔)

Reply:
Here's a PR including the moving of logic: #12090, I can add it into this branch but it will make the diff much hard to read.